paperqa/llms.py

```diff
@@ -622,19 +652,22 @@ async def acomplete_iter(  # type: ignore[override]
     async def achat(  # type: ignore[override]
         self, messages: Iterable[dict[str, str]]
     ) -> Chunk:
-        response = await self.router.acompletion(self.name, messages)
+        response = await self.router.acompletion(self.name, list(messages))
         return Chunk(
-            text=response.choices[0].message.content,
-            prompt_tokens=response.usage.prompt_tokens,
-            completion_tokens=response.usage.completion_tokens,
+            text=cast(litellm.Choices, response.choices[0]).message.content,
+            prompt_tokens=response.usage.prompt_tokens,  # type: ignore[attr-defined]
+            completion_tokens=response.usage.completion_tokens,  # type: ignore[attr-defined]
         )
 
     @rate_limited
     async def achat_iter(  # type: ignore[override]
         self, messages: Iterable[dict[str, str]]
     ) -> AsyncIterable[Chunk]:
         completion = await self.router.acompletion(
-            self.name, messages, stream=True, stream_options={"include_usage": True}
+            self.name,
+            list(messages),
```

Comment:
I think if messages is a generator here, this will pull every value into the list which might not be desired here.

Reply:
The issue is `litellm` types with `list[Message]`, so I was adhering to their type.

Earlier, we didn't type `self.router`, so this snuck under the radar

Reply:
What do you think we should do here?

Reply:
We discussed this in-person